File: custom-addons/morenitapp/models/organizador.py
from odoo import models, fields, api

# ============================================================
# Modelo Organizador
# Representa organizadores de eventos de la hermandad
# ============================================================

class Organizador(models.Model):
    _name = 'morenitapp.organizador'
    _description = 'Organizador de la Hermandad'

    # Campo que se mostrará como nombre del registro
    _rec_name = 'nombre'


    # CIF del organizador
    cif = fields.Char(
        string="CIF",
        required=True
    )

    # Nombre del organizador
    nombre = fields.Char(
        string="Nombre",
        required=True
    )

    # Teléfono de contacto
    telefono = fields.Char(
        string="Teléfono"
    )

    # Correo electrónico
    email = fields.Char(
        string="Email"
    )


    # Calle principal
    calle_id = fields.Many2one(
        'morenitapp.calle',
        string="Calle",
        required=True
    )

    # Localidad obtenida automáticamente desde la calle
    localidad_id = fields.Many2one(
        'morenitapp.localidad',
        related='calle_id.localidad_id',
        string="Localidad",
        store=True,
        readonly=True
    )

    # Código postal automático
    codPostal_id = fields.Many2one(
        'morenitapp.codigopostal',
        related='calle_id.codPostal_id',
        string="C.P.",
        store=True,
        readonly=True
    )

    # Provincia automática
    provincia_id = fields.Many2one(
        'morenitapp.provincia',
        related='localidad_id.codProvincia_id',
        string="Provincia",
        store=True,
        readonly=True
    )

    # Datos adicionales dirección
    puerta = fields.Char(string="Puerta")
    piso = fields.Char(string="Piso")
    numero = fields.Char(string="Número")
    bloque = fields.Char(string="Bloque")
    escalera = fields.Char(string="Escalera")
    portal = fields.Char(string="Portal")

    # ARCHIVOS BINARIOS

    # Logo del organizador
    logo = fields.Binary(
        string="Logo",
        attachment=True
    )

    # Firma digital del presidente
    firma_presidente = fields.Binary(
        string="Firma Presidente",
        attachment=True
    )

    # Firma digital del secretario
    firma_secretario = fields.Binary(
        string="Firma Secretario",
        attachment=True
    )

    # Firma digital del tesorero
    firma_tesorero = fields.Binary(
        string="Firma Tesorero",
        attachment=True
    )

File: mobile_flutter/morenitapp/lib/features/panel-gestion/eventos-cultos/domain/entities/organizador.dart
class Organizador {
  final int id;
  final String cif;
  final String nombre;
  final String? telefono;
  final String? email;

  final int? calleId;
  final String calleName;
  final String numero;
  final String piso;
  final String puerta;
  final String escalera;
  final String bloque;
  final String portal;

  final String? logo;
  final String? firmaPresidente;
  final String? firmaSecretario;
  final String? firmaTesorero;

  Organizador({
    required this.id,
    required this.cif,
    required this.nombre,
    this.telefono,
    this.email,
    this.calleId,
    required this.calleName,
    required this.numero,
    required this.piso,
    required this.puerta,
    required this.escalera,
    required this.bloque,
    required this.portal,
    this.logo,
    this.firmaPresidente,
    this.firmaSecretario,
    this.firmaTesorero,
  });

  factory Organizador.fromJson(Map<String, dynamic> json) {
    return Organizador(
      id: json["id"],
      cif: json["cif"] ?? "",
      nombre: json["nombre"] ?? "",
      telefono: json["telefono"],
      email: json["email"],
      calleId: json["calle_id"],
      calleName: json["calle_name"] ?? "",
      numero: json["numero"] ?? "",
      piso: json["piso"] ?? "",
      puerta: json["puerta"] ?? "",
      escalera: json["escalera"] ?? "",
      bloque: json["bloque"] ?? "",
      portal: json["portal"] ?? "",
      logo: json["logo"],
      firmaPresidente: json["firma_presidente"],
      firmaSecretario: json["firma_secretario"],
      firmaTesorero: json["firma_tesorero"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "cif": cif,
      "nombre": nombre,
      "telefono": telefono,
      "email": email,
      "calle_id": calleId,
      "numero": numero,
      "piso": piso,
      "puerta": puerta,
      "escalera": escalera,
      "bloque": bloque,
      "portal": portal,
      "logo": logo,
      "firma_presidente": firmaPresidente,
      "firma_secretario": firmaSecretario,
      "firma_tesorero": firmaTesorero,
    };
  }
}
